xbmc/cores/AudioEngine/Sinks/AESinkDARWINOSX.cpp

```diff
@@ -93,13 +93,56 @@ static CADeviceList GetDevices()
   return list;
 }
 
+//---------- DEVICE CHANGED EVENT HANDLING
+// when device list change callbacks are received
+// inform the ae engine this amount of time after
+// the last callback was received (bundle multiple callbacks)
+// into one final call
+#define DEVICE_LIST_CHANGED_TIMER_MS 1500
+
+CDeviceChangeListTimeout::CDeviceChangeListTimeout()
+{
+  m_pDeviceListChangedTimer = new CTimer(this);
```

Comment:
One theoretical question though - in your last statement you said:

"if m_deviceChanged and Timer elapsed -> Rescan"

With your approach i don't have a place to do that check ... there is no thread running where i could put that if into - or did you suggest to check for rescan in addpackets of the sink?